datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -239,8 +238,7 @@ pub fn temporal_add_sub_coercion(
             if (is_date(lhs) || is_timestamp(lhs))
                 && (is_date(rhs) || is_timestamp(rhs)) =>
         {
-            // TODO: support Minus
-            None
+            temporal_coercion(lhs, rhs)
```

Comment:
if the comment about error still relevant? 
for `temporal_coercion` if it does the actual substract?